### PR TITLE
[dev] git: enable subpackages deamon, subtree & svn

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak Wani <mwani@microsoft.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adds subpackages to git [daemon, svn & subtree]

###### Change Log  

- Add subpackages

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- localbuild

### CODE REVIEW

File: SPECS/git/git.spec [hunk 1]
```diff
@@ -39,6 +39,61 @@ Requires: git >= 2.1.2
 %description lang
 These are the additional language files of git.
 
+
+%global with_daemon 1
```
Comment (MateuszMalisz) [inline SPECS/git/git.spec:43]: I think we usually declare globals at the top

Comment (mfrw) [reply SPECS/git/git.spec:43]: Sure, i though i would keep them close to where they are used .. but sure .. moving them to a single location.

**Review by MateuszMalisz — APPROVED** [review]
Please apply linter changes before submitting. 🚢

**Comment by mfrw** [comment]
> Please apply linter changes before submitting. 🚢

There was a lot of churn in the spec due to linting .. skipping that for a separate PR :)

**Pull request merged**